File: demo-react-native/src/data/models/task.ts
export enum TaskStatus {
  TODO = 'TODO',
  IN_PROGRESS = 'IN_PROGRESS',
  DONE = 'DONE',
}

export enum TaskPriority {
  LOW = 'LOW',
  MEDIUM = 'MEDIUM',
  HIGH = 'HIGH',
}

export interface Task {
  id: string;
  title: string;
  description: string | null;
  status: TaskStatus;
  priority: TaskPriority;
  createdDate: string | null;
  updatedDate: string | null;
}

export interface TaskRequest {
  title: string;
  description: string | null;
  status: TaskStatus;
  priority: TaskPriority;
}

export interface ErrorResponse {
  message?: string;
}

File: demo-ios/Demo/Data/Model/Task.swift
import Foundation

enum TaskValidation {
    static let maxTitleLength = 100
}

struct Task: Codable, Equatable, Identifiable, Sendable {
    let id: String
    let title: String
    let description: String?
    let status: TaskStatus
    let priority: TaskPriority
    let createdDate: String?
    let updatedDate: String?
}

struct TaskRequest: Codable, Equatable, Sendable {
    let title: String
    let description: String?
    let status: TaskStatus
    let priority: TaskPriority
}

struct ErrorResponse: Codable, Equatable, Sendable {
    let message: String?
}
